nat-lab/tests/test_mesh_plus_vpn.py

```diff
@@ -268,37 +269,39 @@ async def test_mesh_plus_vpn_both_peers(
     [
         pytest.param(
             ConnectionTag.DOCKER_CONE_CLIENT_1,
-            AdapterType.BoringTun,
+            TelioAdapterType.BORING_TUN,
             "10.0.254.1",
         ),
         pytest.param(
             ConnectionTag.DOCKER_CONE_CLIENT_1,
-            AdapterType.LinuxNativeWg,
+            TelioAdapterType.LINUX_NATIVE_TUN,
             "10.0.254.1",
             marks=pytest.mark.linux_native,
         ),
         pytest.param(
             ConnectionTag.WINDOWS_VM_1,
-            AdapterType.WindowsNativeWg,
+            TelioAdapterType.WINDOWS_NATIVE_TUN,
             "10.0.254.7",
             marks=pytest.mark.windows,
         ),
         pytest.param(
             ConnectionTag.WINDOWS_VM_1,
-            AdapterType.WireguardGo,
+            TelioAdapterType.WIREGUARD_GO_TUN,
             "10.0.254.7",
             marks=pytest.mark.windows,
         ),
         pytest.param(
             ConnectionTag.MAC_VM,
-            AdapterType.Default,
+            None,
```

Comment:
What happens if the adapter is set to `None`? Is the default one used for each platform? AFAIK only BoringTun works on macOS

Reply:
The logic for that is handled here: https://github.com/NordSecurity/libtelio/pull/839/files#diff-f1122f50402db2f1613c75eb1db5aed7bc4c9ed907b99f1bf4e5010345bfe6cdR793 and there is a comment for the `SetupParams` class here https://github.com/NordSecurity/libtelio/pull/839/files#diff-9fa2bd5404cf7ee9aeb4e031cc531e8d22f729d73f6891181350bfab552fa181R43 that I updated. Basically, `None` tells natlab to use the default adapter for each platform which is `WireguardGo` on windows and `BoringTun` on all other platforms

Reply:
In that sense the `AdapterType.Default` pattern was a little bit more descriptive of what is actually going on.
What do you think about naming the parameter `override_adapter_type` instead then?